src/main/java/io/split/android/client/service/sseclient/sseclient/NewPushNotificationManager.java

```diff
@@ -0,0 +1,182 @@
+package io.split.android.client.service.sseclient.sseclient;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.annotation.VisibleForTesting;
+
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import io.split.android.client.service.executor.SplitTask;
+import io.split.android.client.service.executor.SplitTaskExecutionInfo;
+import io.split.android.client.service.executor.SplitTaskType;
+import io.split.android.client.service.sseclient.SseJwtToken;
+import io.split.android.client.service.sseclient.feedbackchannel.PushManagerEventBroadcaster;
+import io.split.android.client.service.sseclient.feedbackchannel.PushStatusEvent;
+import io.split.android.client.service.sseclient.feedbackchannel.PushStatusEvent.EventType;
+import io.split.android.client.utils.Logger;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static java.lang.reflect.Modifier.PRIVATE;
+
+public class NewPushNotificationManager {
+
+    private final static int POOL_SIZE = 1;
+    private final static long AWAIT_SHUTDOWN_TIME = 5;
+    private final ScheduledExecutorService mExecutor;
+    private final PushManagerEventBroadcaster mBroadcasterChannel;
+    private final SseAuthenticator mSseAuthenticator;
+    private final NewSseClient mSseClient;
+    private SseRefreshTokenTimer mRefreshTokenTimer;
+    private SseDisconnectionTimer mDisconnectionTimer;
+
+
+    private final static int AUTHENTICATING = 1;
+    private final static int CONNECTING = 3;
```

Comment:
Number 2 is not being used among these status constants. Does it mean that a constant is missing? If not, consider reordering the values (although the logic would be the same)

Reply:
Updated!